### PR TITLE
Migrate over to using EQFORM over FOFORM

### PR DESCRIPTION
**Pull request by ehildenb**
EQFORM is simpler and often all that is needed. FOFORM adds many sorts and complexity.

### CODE REVIEW

**Review by sskeirik — COMMENTED** [review]
Pretty straightforward so not much to address. One issue is that: FOForm, PosEqQFForm, PosEqConj, and EqConj were all changed into type Form. I think the following mapping is more appropriate:
FOForm/QFForm -> Form
PosEqQFForm -> PosEqForm
PosEqConj -> PosEqConj
EqConj -> EqConj
since we have all the sorts anyway (unless they are broken in this particular version, in which case, skip it).

**Comment by ehildenb** [comment]
@skeirik2 I've made the sorts more specific in three places. In `cterms.maude` the functions `#disjSubsts` and `#conjSubst` now produce `PosEqForm` and `PosEqConj` instead of `[Form]`.

In `mtransform.maude`, `unif : Form -> EqCondition` (a way to hold unification problems as side conditions in rules) now takes the more specific `EqConj`.

Unfortunately the second-to-last commit fails tests, as it sends `purificaition.maude` into an infinite loop. I think I uncovered a bug in Maude here with the `print attribute`, as setting `print attribute on` causes the infinite loop to *not* happen. I think we should have the same behavior whether printing is on or off.

**Comment by ehildenb** [comment]
No bug in Maude found, false alarm, just me being a derp.

There was an infinite loop in the purification algorithm which was uncovered by switching to EQFORM. Now the 3rd to last commit provides more exlpicit conditions for the purification algorithm.

**Comment by sskeirik** [comment]
@ehildenb It's not just your problem. The current version of `eqform` has known bugs; hopefully the next version will allow some of the conditions to be removed.

Aside from that, I actually see some potential bugs here. The recursion for `_in_` and `_<<_` both uses variables of sort `Form`; but it looks like in both cases, you want tree traversal and not set traversal. This is the shared disadvantage of both `eqform` and `foform`; naive structural recursion with the topmost sort does not work in general, because the identity operator may appear underneath one of your terms and that can cause an infinite loop. To fix the problem, if it is a problem, for now, you can add conditions on the equations.

EDIT: If it is a problem now, it's probably worth it to add some test cases for those kinds of cases; i.e. singleton formulas; just to make sure the problem gets fixed and stays fixed.

File: contrib/tools/meta.md/eqform.md [hunk 1]
```diff
@@ -238,6 +256,21 @@ fmod EQFORM is
...
+  eq tt << SUB = tt .
+  eq ff << SUB = ff .
+
+  eq (T ?= T') << SUB = (T << SUB) ?= (T' << SUB) .
+  eq (T != T') << SUB = (T << SUB) != (T' << SUB) .
+
+  eq (~ F) << SUB = ~ (F << SUB) .
+
+ ceq (F /\ F') << SUB = (F << SUB) /\ (F' << SUB) if F =/= tt /\ F' =/= tt .
+ ceq (F \/ F') << SUB = (F << SUB) \/ (F' << SUB) if F =/= ff /\ F' =/= ff .
```
Comment (ehildenb) [inline contrib/tools/meta.md/eqform.md:273]: @skeirik2 let's discuss here where it's easier to talk about the equations in question.

I think that the side-conditions are sufficient to assure that we don't have an infinite recursion here?

File: contrib/tools/meta.md/purification.md [hunk 2]
```diff
@@ -201,17 +203,20 @@ fmod PURIFICATION is
...
-    eq (T != T')     in M = wellFormed(M, T) and wellFormed(M, T') .
+    op _in_ : Form Module -> Bool .
+    -------------------------------
+    eq tt        in M = true .
+    eq ff        in M = true .
+    eq (~ F)     in M = F in M .
+   ceq (F /\ F') in M = (F in M) and (F' in M) if F =/= tt /\ F' =/= tt .
+   ceq (F \/ F') in M = (F in M) and (F' in M) if F =/= ff /\ F' =/= ff .
+    eq (T ?= T') in M = wellFormed(M, T) and wellFormed(M, T') .
+    eq (T != T') in M = wellFormed(M, T) and wellFormed(M, T') .
```
Comment (ehildenb) [inline contrib/tools/meta.md/purification.md:219]: I think also here the side conditions are sufficient?

File: contrib/tools/meta.md/eqform.md (hunk 1 above)
Comment (sskeirik) [reply contrib/tools/meta.md/eqform.md:273]: Look good to me.

File: contrib/tools/meta.md/purification.md (hunk 2 above)
Comment (sskeirik) [reply contrib/tools/meta.md/purification.md:219]: This works too.

**Comment by sskeirik** [comment]
@ehildenb Looks good to me.

**Review by sskeirik — APPROVED** [review]

**Pull request merged**